script.js

```diff
@@ -1,4 +1,285 @@
+        cardName = "queen";
+      } else if (cardName == 13) {
+        cardName = "king";
+      }
+
+      var currentValue = rankCounter;
+      if (currentValue == 11 || currentValue == 12 || currentValue == 13) {
+        currentValue = 10;
+      }
+
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        emoji: currentEmoji,
+        blackjackValue: currentValue,
+      };
+      cardDeck.push(card);
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+  return cardDeck;
+};
+
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    var randomIndex = getRandomIndex(cardDeck.length);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    currentIndex = currentIndex + 1;
+  }
+  return cardDeck;
+};
+
+var reset = function () {
+  newDeck = shuffleCards(makeDeck());
+  allCardsDealt = [];
+  playersBlackjackValue = [];
+  playerTurn = 1;
+  gameMode = "draw cards";
+};
+
+var newDeck = shuffleCards(makeDeck());
+var numOfPlayers = 2; // Inclusive of dealer
+var dealer = numOfPlayers; // Dealer is always the last to play
+var allCardsDealt = []; // Holds array of card object arrays
+var playersBlackjackValue = []; // Holds array of final numbers
+var playerTurn = 1; // Default start with player 1
+var gameMode = "draw cards"; // 4 phases - "draw cards", "game", "choice", "results"
+
+// ===== MAIN FUNCTION ===== //
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  //PHASE 1: DRAW CARDS
+  if (gameMode == "draw cards") {
+    // Create object for each player and deal 2 cards each
+    for (var i = 0; i < numOfPlayers; i += 1) {
+      var card1 = newDeck.pop();
+      var card2 = newDeck.pop();
+      var first2Cards = [card1, card2];
+      allCardsDealt.push(first2Cards);
+    }
+    gameMode = "game";
+    myOutputValue += `Drawing <b>first 2 cards</b> 🎲🎲 for each player...`;
+  }
+  //PHASE 2: GAME
+  else if (gameMode == "game") {
+    // Dealer Turn
+    if (playerTurn == dealer) {
+      myOutputValue += `<b><u>Dealer</b></u><br>`;
+      myOutputValue += `Dealer cards are:`;
+      playersBlackjackValue.push("");
+      var dealerCards = "";
+      var currentTotal = 0;
+      var aceCounter = 0;
+      for (var i = 0; i < allCardsDealt[playerTurn - 1].length; i += 1) {
+        dealerCards += `<br>${allCardsDealt[playerTurn - 1][i].name} of ${
+          allCardsDealt[playerTurn - 1][i].emoji
+        }`;
+        if (allCardsDealt[playerTurn - 1][i].name == "ace" && aceCounter == 0) {
+          currentTotal += 11;
+          aceCounter += 1;
+        } else {
+          currentTotal += allCardsDealt[playerTurn - 1][i].blackjackValue;
+        }
+      }
```

Comment:
To make this clearer, these can be made into variable for reuse throughout 
refPlayerCards = allCardsDealt[playerTurn - 1]
refCard = refPlayerCards[i]

If its consistent that the dealer is index 0
refPlayerCards = allCardsDealt[0] 
would be clearer